### ISSUE TITLE
Add HTTPS-enabled grpc support to cluster deployment

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/penumbra#2341]
We have an HTTPS-enabled endpoint for pd's grpc port in the testnet deployment, but it's not working correctly:

```
❯ pcli -n https://grpc.testnet-preview.penumbra.zone view sync
Error: status: Unknown, message: "unable to fetch latest known block height from fullnode: status: Internal, message: \"protocol error: received message with invalid compression flag: 10 (valid flags are 0 and 1) while receiving response with status: 502 Bad Gateway\", details: [], metadata: MetadataMap { headers: {\"content-type\": \"text/html; charset=UTF-8\", \"referrer-policy\": \"no-referrer\", \"content-length\": \"332\", \"date\": \"Tue, 11 Apr 2023 19:05:02 GMT\", \"alt-svc\": \"h3=\\\":443\\\"; ma=2592000,h3-29=\\\":443\\\"; ma=2592000\"} }", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc"} }
```

The problem isn't with pcli; if we run a fullnode with caddy in front of pd (#1887), then pcli is happy:

```
❯ pcli -n https://testnet1.penumbra.zone view sync
Scanning blocks from last sync height 15877 to latest height 16125
[0s] ██████████████████████████████████████████████████     248/248     11212/s ETA: 0s
```

Let's sort out the cluster config so we access pd's grpc service over HTTPS to the cluster. Then we can update the default values for `pcli` to be HTTPS, rather than HTTP.

**Comment by conorsch** [comment]
Roger that, thanks for the heads up. I was prioritizing other work ahead of this ticket, but given the velocity on the HTTPS work overall, will look into this.

**Comment by conorsch** [comment]
Tackling this as my major commitment for Testnet 52. Based on a prior research spike, it appears our use of GCP ingress mandates strict use of HTTP2 for gRPC, meaning we need HTTPS at all layers of the proxy. We can't use the shiny new LE logic for pd, because deploys to previews would exhaust ratelimits quickly. So, instead, I propose we migrate the ingress away from gcp to something like ingress-nginx (which explicitly supports h2c for grpc), or alternatively traefik. I'll prototype that on a devnet, promote to preview, aiming to have it ready and stable for testnet 52.

That'll finally unblock updating the default pcli node value of `http://testnet.penumbra.zone:8080` -> `https://grpc.testnet.penumbra.zone`. Once it's stable, we should also work with Zpoken to update the default value for the web extension.

**Comment by conorsch** [comment]
Didn't make much progress on this ticket for 52, given higher priority on IBC relaying work such as #2330, but aim to pick it up again next sprint. This document explains succinctly that [Traefik has explicit support for `h2c`](https://doc.traefik.io/traefik/user-guides/grpc/), designed to handle our specific use case: we want TLS termination in front of the grpc service, without forcing the grpc service to handle self-signed HTTPS certs on its own—something we've [already decided is undesirable](https://github.com/penumbra-zone/penumbra/issues/1405#issuecomment-1483911688).

**Comment by conorsch** [comment]
> Traefik has explicit support for h2c, designed to handle our specific use case

Spiked on the Traefik as ingress, and it's working very well on a test cluster. The crucial distinction is that the GCP LB we use now has a very strict interpretation of gRPC, requiring HTTP2, and thus requiring HTTPS between every hop. Since we've decided not to add support for manual cert wrangling to `pd`, we need to terminate TLS in front of `pd`, and pass plaintext gRPC traffic to its endpoint. The relevant CRD for Traefik to implement this functionality is quite terse. 

<details>
<summary>
Traefik IngressRoute CRD for gRPC h2c
</summary>

```
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: penumbra-grpc-https
  namespace: penumbra
spec:
  entryPoints:
  - websecure
  routes:
  - kind: Rule
    match: Host(`veil.petrichor.guru`)
    services:
    - name: penumbra
      port: 8080
      scheme: h2c
  tls:
    domains:
    - main: veil.petrichor.guru
    secretName: petrichor.guru-wildcard
```
</details>

That implementation requires a few other components to be installed on the cluster:

* [traefik](https://github.com/traefik/traefik-helm-chart)
* [cert-manager](https://cert-manager.io/docs/installation/helm/)

In the process, we'll be able to remove use the `ManagedCertificate` CRDs and most other GKE-specific tooling, making the config more portable and therefore useful as a reference config. I'm going to work on version controlling the components described above, using a disposable devnet to prototype it. After release of 52, I'll aim to land the changes on preview early in the cycle, to evaluate behavior before using it on a public testnet.

Implementing this will unblock #1888 , and allow us to update the `pcli` default URL, as well.

**Comment by conorsch** [comment]
As of #2598, HTTPS is enabled for pd's gRPC service on preview:

```
❯ pcli -n https://grpc.testnet-preview.penumbra.zone view sync
Scanning blocks from last sync height 0 to latest height 90
[1s] ██████████████████████████████████████████████████      90/90      56/s ETA: 0s
```

The HTTPS URL also works just fine as-is in Buf Studio:

![grpc-https-preview-buf-studio](https://github.com/penumbra-zone/penumbra/assets/657862/6e1f2a20-5627-4518-bf2e-8b920cef4b78)

The historical HTTP-only endpoints are still available for now, too, so using the new endpoint is strictly opt-in:

```
❯ pcli -n http://testnet-preview.penumbra.zone:8080 view sync
Scanning blocks from last sync height 0 to latest height 192
[1s] ██████████████████████████████████████████████████     192/192     147/s ETA: 0s
```

Let's take it for a spin and see if we like it.

**Comment by conorsch** [comment]
Resolved by #2598.
